RequestShare: Introduce the notion of a share request (#194)
cs3org/OCM-API: IETF-RFC.md

* the Receiving Server is unable to act as an API client for (any of) the protocol(s) listed for accessing the resource
* an initial check shows that the resource cannot successfully be accessed through (any of) the protocol(s) listed

# Request for a share
If the Recieving Party knows of a resource that has not yet been shared, the Recieving Party MAY make a HTTP POST to request a share

* to the `/request-share` path in the Sending Server's OCM API
* using `application/json` as the `Content-Type` HTTP request header
* its request body containing a JSON document representing an object with the fields as described below
* using TLS

> * using [httpsig](https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12)

Reviewer: I guess that should be httpsig [RFC9421]

## Fields

> 

Reviewer 2: Don't we also need the sender?
```suggestion
* REQUIRED sender (string)
          OCM Address of the user who will be requested to share the resource.
```

* REQUIRED shareWith (string) 

>           Consumer specific identifier of the user, group or federation that

Reviewer 2: ```suggestion
          OCM Address of the user, group or federation that
```

>           wants to recive a share of the resource.

Reviewer 2: ```suggestion
          wants to receive a share of the resource.
```

        Example: "51dc30ddc473d43a6011e9ebba6ca770@geant.org"

> * REQUIRED shareId (string)

Reviewer 2: Here we have a problem: a receiver won't know the (opaque) ID of a remote resource, but only a path (or something "human readable"?). Shall we name this simply `share`, or `sharePath`?

          A unique identifier for the resource.
        Example: 1234567890abcdef

> After receiving a request for a Share, the Sending Party MAY send a Share Creation Notification to the Sending Party using the OCM address in the shareWith field.

Reviewer 2: Here I'd imagine the flow to be async: the server would respond HTTP 202 Accepted, signifying that the `sender` user was notified about the request to share. Once they accept, the normal Share Notification will take place.
